### PR TITLE
Update shader warm-up for recent Skia changes

### PR DESCRIPTION
**Pull request by liyuqian**
The update is copied from an update we made to a Google-internal
client: cl/260202900

The update will save 1 shader compilation.

This should help solve our regression:
https://github.com/flutter/flutter/issues/31203

More regressions on iOS might be introduced later by
https://github.com/flutter/engine/pull/9813#issuecomment-520039890

Unfortunately, we didn't rebase our benchmarks so such regressions
were not detected.

Hence to fully solve https://github.com/flutter/flutter/issues/31203,
we might need to revert some change in
https://github.com/flutter/engine/pull/9813 to make iOS shader warm-up
happen on the GPU thread again.

### CODE REVIEW

File: packages/flutter/lib/src/painting/shader_warm_up.dart [hunk 1]
```diff
@@ -192,5 +192,22 @@ class DefaultShaderWarmUp extends ShaderWarmUp {
...
+    for (double fraction in <double>[0.0, 0.5]) {
+      canvas
+        ..save()
+        ..translate(fraction, fraction)
+        ..clipRRect(ui.RRect.fromLTRBR(8, 8, 328, 248, const ui.Radius.circular(16)))
+        ..drawRect(const ui.Rect.fromLTRB(10, 10, 320, 240), ui.Paint())
+        ..restore();
+      canvas.translate(drawCallSpacing, 0.0);
+    }
+    canvas.translate(0.0, drawCallSpacing);
```
Comment (dnfield) [inline packages/flutter/lib/src/painting/shader_warm_up.dart]: Can we add a test for this with a TestCanvas that just captures the method invocations and verifies they come out to what we expect?  If you grep for TestCanvas you should find some examples

Comment (liyuqian) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: How about the added test?

Comment (dnfield) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: Thanks!  I'll defer to other reviewer(s) on final review for this.

**Comment by liyuqian** [comment]
Ping @chinmaygarde @jason-simmons @cbracken

**Comment by liyuqian** [comment]
@iskakaushik : since you're interested in shader warm-up, I wonder if you'd like to review this PR? For reference, go/flutter-compile-janks is an work-in-progress design doc that includes some backgrounds and solution comparisons. I'm also happy to give  a brown bag if you're interested 😄

**Comment by liyuqian** [comment]
Ping...

**Comment by liyuqian** [comment]
Ping...

**Review by iskakaushik — APPROVED** [review]
LGTM

**Pull request merged**